src/utility/copy.cpp

```diff
@@ -81,6 +82,18 @@ appendCreateRelationCopyString(StringInfo info, ParseState *pstate, CopyStmt *co
 
 		appendStringInfoString(info, quote_identifier(strVal(attr)));
 	}
+#else
```

Comment:
This `#if/#else` shouldn't be necessary afaict. These pieces of code should have the same behaviour.

Reply:
PG14 does not have a String structure, so foreach_node cannot be used.
[https://github.com/postgres/postgres/blob/REL_14_STABLE/src/include/nodes/value.h](https://github.com/postgres/postgres/blob/REL_14_STABLE/src/include/nodes/value.h)
[https://github.com/postgres/postgres/blob/REL_15_STABLE/src/include/nodes/value.h](https://github.com/postgres/postgres/blob/REL_15_STABLE/src/include/nodes/value.h)

Reply:
Hmm okay, now I understand. Can you then change it to something like this:

```c
#if PG_VERSION_NUM >= 150000
	foreach_node(String, attr, copy_stmt->attlist) {
# else
	foreach_ptr(Value, attr, copy_stmt->attlist) {
#endif
```

Reply:
Wow, that's a good idea.